### PR TITLE
Issue 20986 key based time expire cache provider

### LINKED ISSUE #20986: Create cache provider for key-based time expiration

### PR DESCRIPTION
**Pull request by dsilvam**
* Modifies Caffeine Cache to support an Expirable object that provides a TTL for key-specific time expiration of the entry in cache. 
* Makes GraphQLCache use this new ability instead of implementing it itself.

### CODE REVIEW

**Review by nollymar — APPROVED** [review]

**Review by fabrizzio-dotCMS — COMMENTED** [review]
I thought we were gonna Update ExpiringMap class to be able to support different ttls using caffeine built-in functionality. And Reuse the ExpiringMap as the de facto implementation for expiring caches

**Comment by dsilvam** [comment]
@fabrizzio-dotCMS we can do that in a separate PR

File: dotCMS/src/main/java/com/dotcms/graphql/GraphQLCache.java [hunk 1]
```diff
@@ -46,32 +45,19 @@
...
-                    remove(key);
-                }
-            }
-        }
+        final Object value = cache.getNoThrow(cacheKey, getPrimaryGroup());
+
+        refreshKey(cacheKey, valueSupplier, ttl);
 
-        return result;
+        return value instanceof ExpirableCacheEntry
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/GraphQLCache.java:58]: shpuld it be Expirable? Also I think is better use the Expirable.class.isInstance()

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/GraphQLCache.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/GraphQLCache.java:58]: @freddyucv ExpirableCacheEntry is ok - Expirable doesn't have the "getResults" method, only the getTTL

**Review by freddyucv — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]
very nice

File: dotCMS/dependencies.gradle [hunk 2]
```diff
@@ -409,7 +409,7 @@ dependencies {
     compile group: 'com.twelvemonkeys.servlet', name: 'servlet', version:'3.7.0'
     
     
-    compile 'com.github.ben-manes.caffeine:caffeine:2.4.0'
+    compile 'com.github.ben-manes.caffeine:caffeine:2.5.0'
```
Comment (ben-manes) [inline dotCMS/dependencies.gradle:412]: We follow semver strictly. Please upgrade to the latest 2.x to minimize bugs and have the best experience.

Comment (dsilvam) [reply dotCMS/dependencies.gradle:412]: Sure thanks @ben-manes

**Pull request merged**